src/backend/marsha/core/tests/test_api_timed_text_track.py

```diff
@@ -554,83 +917,250 @@ def test_api_timed_text_track_create_staff_or_user(self):
+        self.assertEqual(TimedTextTrack.objects.count(), 0)
+
+    def test_api_timed_text_track_create_by_video_organization_admin(self):
+        """
+        Organization administrator token user creates a timed text track for a video.
+
+        A user with a user token, who is an organization administrator, can create a timed
+        text track for a video that belongs to that organization.
+        """
+        user = factories.UserFactory()
+        # A playlist where the user is an instructor, with a video
+        organization = factories.OrganizationFactory()
+        playlist = factories.PlaylistFactory(organization=organization)
+        video = factories.VideoFactory(playlist=playlist)
+        factories.OrganizationAccessFactory(
+            user=user, organization=organization, role=models.ADMINISTRATOR
+        )
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(user.id)
+        jwt_token.payload["user"] = {"id": str(user.id)}
+
+        response = self.client.post(
+            "/api/timedtexttracks/",
+            {"language": "fr", "video": str(video.id)},
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(TimedTextTrack.objects.count(), 1)
+        content = json.loads(response.content)
+        self.assertEqual(
+            content,
+            {
+                "id": str(TimedTextTrack.objects.first().id),
+                "active_stamp": None,
+                "is_ready_to_show": False,
+                "mode": "st",
+                "language": "fr",
+                "upload_state": "pending",
+                "source_url": None,
+                "url": None,
+                "video": str(video.id),
+            },
+        )
+
+    def test_api_timed_text_track_update_detail_anonymous(self):
+        """Anonymous users should not be allowed to update a timed_text_track through the API."""
+        timed_text_track = TimedTextTrackFactory(language="fr")
+        data = {"language": "en"}
+        response = self.client.put(
+            f"/api/timedtexttracks/{timed_text_track.id}/",
+            json.dumps(data),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 401)
+        timed_text_track.refresh_from_db()
+        self.assertEqual(timed_text_track.language, "fr")
+
+    def test_api_timed_text_track_update_detail_token_user_language(self):
+        """Token users should be able to update the language of their timed_text_track."""
+        timed_text_track = TimedTextTrackFactory(language="fr")
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(timed_text_track.video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+
+        data = {"language": "en"}
+        response = self.client.put(
```

Comment:
In this test you are getting the resource before updating it like in other test testing the `PUT` verb